### PR TITLE
Add /me endpoint to user api.

### PR DESCRIPTION
**Pull request by andy9775**
/me returns the current user if authenticated and exists.

### CODE REVIEW

File: server/routes/v1/user.ts [hunk 1]
```diff
@@ -129,11 +130,21 @@ async function resetPassHandler(req: Request, res: Response) {
   }
 }
 
+async function me(req: Request, res: Response) {
+  try {
+    const user = getAuthUser(res);
+    res.status(200).json(user);
+  } catch (err) {
+    console.error(err);
```
Comment (plausibly) [inline server/routes/v1/user.ts:138]: Respond with an error message/unauthorized?

**Review by plausibly — CHANGES_REQUESTED** [review]

File: server/routes/v1/user.ts (hunk 1 above)
Comment (andy9775) [reply server/routes/v1/user.ts:138]: `getAuthUser` does.
https://github.com/GDSCUTM-CommunityProjects/UBoard/blob/main/server/middleware/auth.ts#L23

**Review by danielduong8 — APPROVED** [review]

File: server/routes/v1/user.ts (hunk 1 above)
Comment (plausibly) [reply server/routes/v1/user.ts:138]: Should we have any tests for this, since there doesnt seem to be any for getauthuser? Other than that it looks good

**Review by plausibly — APPROVED** [review]

File: server/routes/v1/user.ts (hunk 1 above)
Comment (andy9775) [reply server/routes/v1/user.ts:138]: Really this type of a test would be: 1/ set the user on locals 2/ assert that it was set (getAuthUser returns the user). So I don't really think it's worth testing a setter/getter. LMKWYT!

It is Indirectly tested in other places as well.

Comment (plausibly) [reply server/routes/v1/user.ts:138]: Fair enough. I think it's good to merge then.

**Review by plausibly — CHANGES_REQUESTED** [review]
I think this needs to go through the auth middleware before calling me

File: server/routes/v1/user.ts [hunk 2]
```diff
@@ -129,11 +130,21 @@ async function resetPassHandler(req: Request, res: Response) {
...
+  }
+}
+
 userRouter.post("/signin", signInHandler);
 userRouter.post("/signup", signUpHandler);
 userRouter.post("/signout", signOut);
 
 userRouter.get("/confirm", confirmEmailHandler);
 userRouter.get("/password-reset", resetPassHandler);
+userRouter.get("/me", me);
```
Comment (plausibly) [inline server/routes/v1/user.ts:148]: I think this needs to go through the auth middleware before calling me

**Review by plausibly — CHANGES_REQUESTED** [review]

File: server/routes/v1/user.ts (hunk 2 above)
Comment (danielduong8) [reply server/routes/v1/user.ts:148]: ^

Comment (andy9775) [reply server/routes/v1/user.ts:148]: #42

**Review by plausibly — APPROVED** [review]

**Pull request merged**